dynamo_db_base_block.py

```diff
@@ -64,29 +98,34 @@ def process_signals(self, signals, input_id='default'):
                 batch_groups[table_name].append(sig)
             except:
                 self._logger.exception("Unable to add signal to table list")
-
         self._logger.debug("Processing {} signals in {} batch groups".format(
             len(signals), len(batch_groups)))
+        return batch_groups
 
-        # Go through each table in batch groups and batch write all of the
-        # signals to the table
-        for table_name, sigs in batch_groups.items():
-            self._logger.debug("Saving {} signals to {}".format(
-                len(sigs), table_name))
-            try:
-                # Lock around each table - in case it is creating still
-                self._logger.debug(
-                    "Waiting for table lock on {}".format(table_name))
-                with self._table_locks[table_name]:
-                    self._logger.debug(
-                        "Table lock acquired for {}".format(table_name))
-                    table = self._get_table(table_name)
-                    self._save_signals_to_table(table, sigs)
-            except:
-                self._logger.exception("Could not batch write to table {}"
-                                       .format(table_name))
+    def _process_table_signals(self, table_name, signals):
+        """ Go through each table in batch groups and batch operate all of the
+        signals to the table
 
-    def _get_table(self, table_name):
+        Returns:
+            signals(list): Any signals to notify
+        """
+        output = []
+        # Lock around each table - in case it is creating still
+        self._logger.debug(
+            "Waiting for table lock on {}".format(table_name))
+        with self._table_locks[table_name]:
+            self._logger.debug(
+                "Table lock acquired for {}".format(table_name))
+            table = self._get_table(table_name)
+            if table:
+                out_sigs = self.execute_signals_query(table, signals)
+                if out_sigs:
+                    output.extend(out_sigs)
+            else:
+                self._logger.error("Could not get table")
```

Comment:
I'd be ok raising an exception here, rather than just logging. The caller will catch it